agent/bench-scripts/pbench-fio

```diff
@@ -601,7 +601,7 @@ function fio_run_job() {
 			if pbench-is-local ${client}; then
 				debug_log "killing any old local fio processes and starting new a fio process for the local client"
 				killall ${benchmark} > /dev/null 2>&1
-				screen -dmS fio-server bash -c '${benchmark_bin} --server=,${client_ports[${client}]} > client-result.txt 2> client-result.err'
+				screen -dmS fio-server bash -c "${benchmark_bin} --server=,${client_ports[${client}]} > '$benchmark_results_dir/client-result.txt' 2> '$benchmark_results_dir/client-result.err'"
```

Comment:
I'm a bit unclear on how `screen` handles environment variables: but isn't it better to expand the `$benchmark_results_dir` here rather than deferring it into the `bash` under `screen`?? Can it be different? (Plus we usually prefer `${benchmark_results_dir}` just for clarity even in contexts like this where it doesn't matter.)

Reply:
That is exactly what I did. Previously the single quotes were used, therefore it tried to get variables from the screen session (which lacked the ones that were not exported like ports). Now with using double quotes the variables are replaced before executing screen and the single quotes are there to ensure the expanded variables won't be interpreted by screen as multiple commands.

As for the `${}`, I'll change that in v2 (can't get used to this style...)

Reply:
@dbutenhof do you know why the jenkins/pr-merge check failed? I tried looking at the logs but the output changed and I'm not sure what to grep for...

Reply:
OK, it was still running, I can see the output now... I'll add the `${}` and try to fix the selftests probably tomorrow.

Reply:
> That is exactly what I (hope I) did. Previously the single quotes were used, therefore it tried to get variables from the screen session

Sure; but you *added* `'$benchmark_results_dir/client-result.txt'`, so the single quotes ought to prevent substitution, if I'm reading it right. I think the sub-bash under `screen` will see these *un*quoted, and will *probably* inherit the environment variable from `pbench-fio`, so I expect it'll work out fine ... but doing the substitution on this side is possibly better.

Reply:
First the double quotes are being used, therefore bash expands all the "$" variables and passes it expanded to the screen process including the single quotes. So an example:

```
foo="file name with spaces"
touch $foo
# creates 4 files
touch "$foo"
# creates the correct file
touch "'$foo'"
# creates a single file but the name contains extra quotes
bash -c touch "$foo"
# fails as the "file name with spaces" is an extra argument
bash -c "touch $foo"
# creates 4 files
bash -c "touch '$foo'"
# creates the correct file, because before passing the arguments to the to-be-executed bash expands the variables, so it results in `["bash", "-c", "touch 'file name with spaces"]` execution. The executed bash than expands the single quotes and the execution becomes `["touch", "file name with spaces"]`.
```

This is universal in bash, not tied to screen.

Reply:
Interesting. The nested quoting is handled differently than I'd expected. Thanks.